config/confighttp/confighttp.go

```diff
@@ -156,11 +154,11 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 	}
 
 	if hcs.Auth != nil {
-		if ext == nil {
+		if host.GetExtensions() == nil {
```

Comment:
Instead of calling `GetExtensions()` multiple times, just add a line `ext := host.GetExtensions()`

Reply:
Done